dlt/extract/hints.py

```diff
@@ -40,10 +38,66 @@ class TResourceHints(TypedDict, total=False):
     merge_key: TTableHintTemplate[TColumnNames]
     incremental: Incremental[Any]
     schema_contract: TTableHintTemplate[TSchemaContract]
+    table_format: TTableHintTemplate[TTableFormat]
     validator: ValidateItem
     original_columns: TTableHintTemplate[TAnySchemaColumns]
 
 
+class HintsMeta:
+    __slots__ = "hints"
```

Comment:
i never heard of this before, this is purely here for performance reasons, right?